### PR TITLE
[Schema Registry API reference] Support `/mode` for Schema Registry 

### PR DESCRIPTION
**Pull request by kbatuigas**
## Description

Resolves https://github.com/redpanda-data/documentation-private/issues/2619
Resolves https://github.com/redpanda-data/documentation-private/issues/2526
Review deadline: 30 July 2024

Related to https://github.com/redpanda-data/docs/pull/647

## Page previews

[`GET /mode`](https://deploy-preview-646--redpanda-docs-preview.netlify.app/api/pandaproxy-schema-registry/#get-/mode)
`PUT /mode`
`GET /mode/{subject}`
`PUT /mode/{subject}`
`DELETE /mode/{subject}`

## Checks

- [x] New feature
- [ ] Content gap
- [ ] Support Follow-up
- [ ] Small fix (typos, links, copyedits, etc)

### CODE REVIEW

**Review by Feediver1 — APPROVED** [review]
lgtm
Thank you @kbatuigas !

**Review by mattschumpert — APPROVED** [review]

File: modules/ROOT/attachments/admin-api.yaml [hunk 1]
```diff
@@ -851,6 +851,17 @@ paths:
         200:
           description: Success
           content: {}
+  /partitions/rebalance_cores:
```
Comment (kbatuigas) [inline modules/ROOT/attachments/admin-api.yaml:854]: This belongs to https://github.com/redpanda-data/docs/pull/615/commits which should be merged by the GA release

File: modules/ROOT/attachments/admin-api.yaml
Comment (kbatuigas) [inline modules/ROOT/attachments/admin-api.yaml:1]: The changes to this file belong to https://github.com/redpanda-data/docs/pull/615/commits and will be included in the GA release

**Review by BenPope — APPROVED** [review]
I pushed a commit that synchronises this with the latest swagger published by schema registry (https://github.com/redpanda-data/redpanda/pull/22665)

**Pull request merged**